shape.c

```diff
@@ -430,7 +486,10 @@ rb_shape_transition_shape_capa_create(rb_shape_t* shape, uint32_t new_capacity)
 rb_shape_t *
 rb_shape_transition_shape_capa(rb_shape_t* shape)
 {
-    return rb_shape_transition_shape_capa_create(shape, shape->capacity * 2);
+    // Grow to a power of 2
+    uint32_t new_capacity = 1 << (32 - nlz_int32((shape->capacity * 2) - 1));
+    new_capacity -= slots_padding;
```

Comment:
I wonder if I should move this to `gc.c` as it could be used elsewhere, e.g. for growing `RArray`.